src/hooks/useStakingSlider.ts

```diff
@@ -53,6 +57,29 @@ const useStakingSlider = (isObjection: boolean) => {
     userMinStake,
   );
 
+  const { data, loading: loadingReputation } = useGetUserReputationQuery({
+    variables: {
+      input: {
+        colonyAddress: actionData.getColonyAction?.colonyAddress ?? '',
+        walletAddress: user?.walletAddress ?? '',
+        domainId: Number(motionDomainId),
+        rootHash,
+      },
+    },
+  });
+
+  const userReputation = data?.getUserReputation;
+  /* User cannot stake more than their reputation in tokens. */
+  const userMaxStake = BigNumber.from(userReputation ?? '0');
```

Comment:
Cannot see a simple way of reusing the logic to get `userMaxStake`. I would wrap it up into a hook but I wouldn't be able to access colony or staking widget context inside, and it would therefore require an awkward number of props as args. 

I think duplication is warranted here.

Reply:
I mean, it repeats like 2 times only. So you really shouldn't worry about. It's a good compromise